woocommerce/google-listings-and-ads
# Add/1165 split account product issues

### Changes proposed in this Pull Request:

Closes part of #1165 

One of the requirements in the new GMC Account review request is to split the issues between Products issues and Account type issues. This PR addresses that.

More info -> pcTzPl-bx-p2
https://www.figma.com/file/eQ9O4m2flzrcAiDMXkOW0m/Google-Listings-%26-Ads-v1.x?node-id=3736%3A72890

### Screenshots:

![Screenshot 2022-01-03 at 13 47 10](https://user-images.githubusercontent.com/5908855/147932241-081bbef6-f8d1-46b4-99c9-1f61497fe5ab.png)


Case 1 - Multiple account and products (pagination)

https://user-images.githubusercontent.com/5908855/147931779-b6dc2d97-3587-4209-ab88-cfb61e6d893d.mov

https://user-images.githubusercontent.com/5908855/147931936-c504e9be-3988-4573-a46b-2c67fa9ae17d.mov

Case 2 - Only account/Products

https://user-images.githubusercontent.com/5908855/147931920-acfa7e1b-a797-4746-96f6-068d6289a89f.mov

Case is 4 - No issues

https://user-images.githubusercontent.com/5908855/147931951-c8767cfa-9fc5-4f0e-beff-16736710d51d.mov


### Detailed test instructions:


**Test issues table, navigation and switcher:**

Case 1:
- Open the database and go to `wp_gla_merchant_issues` table
-  Create 10 account issues and 13 product issues (if you change the number just be sure you test that numbers makes sense in the totals and you tested the paginator as well)

_Testing procedure:_

1. Go to Product feed
2. It loads 
3. You can see the total of issues in the Status section at the top
4. The Issues block loads
5. You can see 2 sections: Account and Product, with a number between parenthesis showing the total of issues
6. By default the Account type is selected
7. Account shows 5 account issues
8. Go to the next page with the paginator
9. On loading it shows a Placeholder table. It should show page 2 with other 5 account issues after loading.
10. Switch to Product issues
11. It is in page 1 showing 5 product issues
12. Move to Account again
13. It shows page 2 with 5 account issues
14. Move to product again
15. It shows still page 1 with 5 product issues 
16. Move to page 2 in Products. 
17. On loading it shows a Placeholder table. It should show another 5 issues after loading
18. Move to page 3 in Products
19. On loading it shows a Placeholder table. It should show another 5 issues after loading
20. Go back to Account
21. It should show page 2 with 5 account issues
22. Go back to Product.
23. Go to the first page in Products. No placeholders are shown. 
24. You can navigate correctly.
25. Go to Account, it still in page page 2 with 5 account issues


Case 2:
- Open the database and go to `wp_gla_merchant_issues` table
- Remove all the Product issues , in account left only 3 issues. 

_Testing procedure:_

1. Go to Product feed
2. It loads 
3. You can see the total of issues in the Status section at the top
4. The Issues block loads
5. You can see 2 sections: Account and Product, with a number between parenthesis showing the total of issues. In product it's (0).
6. By default the Account type is selected
7. Account shows 3 account issues and no Paginator 
8. Switch to Product issues
9. It shows a table with a message and no product issues
10. Switch back to Account


Case 3:
- Open the database and go to `wp_gla_merchant_issues` table
- Remove all the Account issues , in Product create only 3 issues. 

_Testing procedure:_

1. Go to Product feed
2. It loads 
3. You can see the total of issues in the Status section at the top
4. The Issues block loads
5. You can see 2 sections: Account and Product, with a number between parenthesis showing the total of issues. In Account it's (0).
6. By default the Account type is selected
7. Account shows a table with a message and no product issues
8. Switch to Product issues
9. It shows a table 3 Product issues
10. Switch back to Account

Case 4:
- Open the database and go to `wp_gla_merchant_issues` table
- Remove all the Account  issues and Product issues as well.  

_Testing procedure:_

1. Go to Product feed
2. It loads 
3. You can see the total of issues in the Status section at the top
4. The Issues block doesn't load

Testing URL query:

1. Go to `/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fproduct-feed`
2. Account issue type is selected
3. Select products
4. See how the URL added  the `issueType=product` query
5. Go to `/wp-admin/admin.php?page=wc-admin&issueType=account&path=%2Fgoogle%2Fproduct-feed`
6. Account is selected
7. Go to `/wp-admin/admin.php?page=wc-admin&issueType=product&path=%2Fgoogle%2Fproduct-feed`
8. Product is selected
9. Select account
10.  See how the URL added  the `issueType=account` query


### Changelog entry

> Add - Split Issues between Product and Account Issues
